src/main/java/com/shopping_app/shoppingApp/service/UserService.java

```diff
@@ -0,0 +1,118 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final AddressRepository addressRepository;
+    private final CartRepository cartRepository;
+    private final PasswordEncoder bCryptPasswordEncoder;
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenProvider tokenProvider;
+
+    public UserResponse registerUser(UserRegisterRequest userRegisterRequest) {
+        Optional<User> existingUser = userRepository.findByEmail(userRegisterRequest.getEmail());
+        if (existingUser.isPresent()) {
+            throw new UserAlreadyExist("User with this email Already exist", HttpStatus.BAD_REQUEST);
+        }
+        User user = convertToUserDomain(userRegisterRequest);
+        user.setRoleName(UserRole.CUSTOMER);
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+        User savedUser = userRepository.save(user);
+        if (null != userRegisterRequest.getAddress()) {
+            Set<Address> address = userRegisterRequest.getAddress().stream().map(AddressService::convertToAddress).collect(Collectors.toSet());
+            setAddressToUser(address, savedUser);
+            savedUser.setAddress(address);
+        }
+        // After registration create cart for users
+        createEmptyCart(savedUser);
+        return UserResponse.from(savedUser);
+    }
+
+    public UserLoginResponse loginUser(UserLoginRequest userLoginRequest) {
+        Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(userLoginRequest.getEmail(), userLoginRequest.getPassword()));
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        String token = tokenProvider.createToken(userLoginRequest.getEmail());
+        return UserLoginResponse.from(userLoginRequest, token);
+    }
+
+    public List<UserResponse> getAllUsers() {
+        List<User> users = userRepository.findAll();
+        return users.stream().map(UserResponse::from).collect(Collectors.toList());
+    }
+
+    @Transactional
```

Comment:
any specific reason to annotate this method with `@Transactional`?

Reply:
Yeah have faced a issue when fetching an user ,  
which is db session is getting closed before fetching the address of user 

solution found : 
  1.   make eager fetch 
  2 . make transctional.